app/models/loan.rb

```diff
@@ -211,4 +215,34 @@ def healthy?
   def health_status_available?
     return !health_check.nil?
   end
+
+  def sum_of_disbursements(start_date: nil, end_date: nil)
+    return nil if transactions.empty?
+    transactions.by_type("disbursement").in_date_range(start_date, end_date).map { |t| t.amount }.sum
+  end
+
+  def sum_of_repayments(start_date: nil, end_date: nil)
+    return nil if transactions.empty?
+    transactions.by_type("repayment").in_date_range(start_date, end_date).map { |t| t.amount }.sum
+  end
+
+  def change_in_interest(start_date: nil, end_date: nil)
+    return nil if transactions.empty?
+    changes = transactions.in_date_range(start_date, end_date).map { |t| (t.change_in_interest) }
+    if changes.any? { |i| i.nil? }
+      raise Accounting::TransactionDataMissingError
```

Comment:
Without a custom error, what the user sees is many loans with `"NoMethodError: no method '+' for nil.` This is just an attempt to make it clear that that error should be fixed by updating all loans. Happy for suggestions on better ways to do this.

Reply:
`raise Account::TransactionDataMissingError if changes.any(&:blank?)` would shorten this up a bit